cudamat/cudamat_kernels.cu

```diff
@@ -781,3 +781,24 @@ __global__ void kWhere(float* condition_mat, float* if_mat, float* else_mat, flo
         target[i] = condition_mat[i] ? if_mat[i] : else_mat[i];
     }
 }
+
+
+__global__ void kCorrelate(float* source, float* kernel, float* dest, int width, int height, int fwidth, int fheight) {
```

Comment:
If it doesn't bother you, it'd be good to also rename `fwidth` and `fheight` to `kwidth` and `kheight`. Not a big deal though.